types/query.d.ts

```diff
@@ -237,7 +252,7 @@ declare module 'mongoose' {
     allowDiskUse(value: boolean): this;
 
     /** Specifies arguments for an `$and` condition. */
-    and(array: FilterQuery<RawDocType>[]): this;
+    and(array: FilterQuery<DocType>[]): this;
```

Comment:
Why changing from `RawDocType` to `DocType` here? I think `RawDocType` is more correct to avoid filtering by document properties like `$isNew`.

Reply:
The reason for changing this was because some test cases like the following `autoTypedQuery()` were failing w.r.t. types:
```
function autoTypedQuery() {
  const AutoTypedModel = autoTypedModel();
  const query = AutoTypedModel.find();
  expectType<typeof query>(AutoTypedModel.find().byUserName('')); // This was throwing type error
}
```

Now, the reason these function were failing because in `models.d.ts`, a function like `find()` was defined like this:
```
find<ResultDoc = THydratedDocumentType>(
    ): QueryWithHelpers<Array<ResultDoc>, ResultDoc, TQueryHelpers, TRawDocType, 'find'>;
```

Notice how in the above declared `find()` type, DocType = ResultDoc and RawDocType = TRawDocType for the corresponding `Query` type which will be generated by the `QueryWithHelpers` type. This also means that the corresponding filters type for that query will be `FilterQuery<TRawDocType>`

**BUT**, in the above autoTypedTest's `byUserName()` method, the `this.query` context being used inside that method has the type `Query` with both DocType and RawDocType being set as the THydratedDocumentType due to auto typing. Thus, when you call the `byUserMethod`, it expects filters of type `FilterQuery<ResultDoc>` instead of `FilterQuery<TRawDocType>`, thus creating a type mismatch error.

To prevent the above type conflict, I chose to replace the `RawDocType` with just `DocType`